### ISSUE TITLE
getStaticProps incompatible with hybrid AMP dynamic routes

### ISSUE THREAD

**Issue by filipposarzana** [vercel/next.js#17935]
# Bug report
## Describe the bug
`getStaticProps` receive different params when using AMP `hybrid` and dynamic routes.

## To Reproduce
Steps to reproduce the behavior, please provide code snippets or a repository:
1 - Create an hybrid AMP page with dynamic route (for example `pages/[...fragments].js`)
2 - add `getStaticPaths` and return data (`{params: { fragments: ['foo', 'bar'] }}`)
2 - Run `npm run dev`
3 - add `getStaticProps` and console.log(params)
4 - you will get for one page `fragments: ['foo', 'bar']` and then `{ amp: '1' }`

## Expected behavior
`getStaticProps` should receive all dynamic params and in addition `amp`

## Code Sample
```typescript
export const config = {
  amp: 'hybrid',
}

export const getStaticPaths = async () => {
  return {
    fallback: false,
    paths: [{
		params: {
			fragments: ['foo', 'bar']
		},
	}],
  }
}

export const getStaticProps = async ({ params }: Param) => {
  console.log(params)

  return {
    props: {
      fragments: params.fragments,
    },
  }
}
```

## System information
* Version of Next.js: `9.5.5`

## Additional context
Somewhat really similar to:
- https://github.com/vercel/next.js/issues/11446
- https://github.com/vercel/next.js/pull/17461

**Comment by timneutkens** [comment]
Can you check `next@canary`, we merged a fix yesterday.

**Comment by filipposarzana** [comment]
Confirm, it's working 🚀 
You guys are amazing!

**Comment by timneutkens** [comment]
Will close this issue then as it's already on canary 👍

**Comment by filipposarzana** [comment]
@timneutkens It's happening again since I updated to latest canary.
It prints `{ amp: 1 }` only once 🤷

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
